code/modules/antagonists/voidwalker/voidwalker_organs.dm

```diff
@@ -0,0 +1,148 @@
+/// Voidwalker eyes with nightvision and thermals
+/obj/item/organ/internal/eyes/voidwalker
+	name = "blackened orbs"
+	desc = "These orbs will withstand the light of the sun, yet still see within the darkest voids."
+	eye_icon_state = null
+	pepperspray_protect = TRUE
+	flash_protect = FLASH_PROTECTION_WELDER
+	color_cutoffs = list(20, 10, 40)
+	sight_flags = SEE_MOBS
+
+/// Voidwalker brain stacked with a lot of the abilities
+/obj/item/organ/internal/brain/voidwalker
+	name = "cosmic brain"
+	desc = "A mind fully integrated into the cosmic thread."
+	icon = 'icons/obj/medical/organs/shadow_organs.dmi'
+
+	/// Alpha we have in space
+	var/space_alpha = 30
+	/// Alpha we have elsewhere
+	var/non_space_alpha = 255
+	/// We settle the un
+	var/datum/action/unsettle = /datum/action/cooldown/spell/pointed/unsettle
+	/// Regen effect we have in space
+	var/datum/status_effect/regen = /datum/status_effect/shadow_regeneration
+	/// The void eater armblade
+	var/obj/item/glass_breaker = /obj/item/void_eater
+
+/obj/item/organ/internal/brain/voidwalker/on_mob_insert(mob/living/carbon/organ_owner, special, movement_flags)
+	. = ..()
+
+	RegisterSignal(organ_owner, COMSIG_ATOM_ENTERING, PROC_REF(on_atom_entering))
+	organ_owner.remove_from_all_data_huds()
+
+	organ_owner.AddComponent(/datum/component/space_camo, space_alpha, non_space_alpha, 2 SECONDS)
+
+	unsettle = new unsettle ()
```

Comment:
```suggestion
	unsettle = new unsettle(organ_owner)
```
actions take a target as the first arg which they track to make sure the action is cleaned up at the right time, important for memory management

alternately maybe you could do `unsettle = new unsettle(src)` on init and just reassign the action around? unsure if that works or is smart